BParkingNano/plugins/TrgBitTableProducer.cc

```diff
@@ -0,0 +1,114 @@
+//// table to produce hlt bits that we are going to use in the analysis
+
+
+// system include files
+#include <memory>
+
+
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "DataFormats/Common/interface/View.h"
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "DataFormats/NanoAOD/interface/FlatTable.h"
+#include "FWCore/Common/interface/TriggerNames.h"
+#include "DataFormats/HLTReco/interface/TriggerEvent.h"
+#include "DataFormats/HLTReco/interface/TriggerObject.h"
+#include "HLTrigger/HLTcore/interface/defaultModuleLabel.h"
+#include "DataFormats/HLTReco/interface/TriggerFilterObjectWithRefs.h"
+#include "DataFormats/PatCandidates/interface/TriggerObjectStandAlone.h"
+#include "TString.h"
+#include <string>
+
+
+
+class TrgBitTableProducer : public edm::stream::EDProducer<> {
+
+
+public:
+
+ 
+ explicit TrgBitTableProducer(const edm::ParameterSet &cfg):
+    trgresultsToken_(consumes<edm::TriggerResults>(cfg.getParameter<edm::InputTag> ("triggerresults"))),
+    hltpaths_( cfg.getParameter< std::vector<std::string> >( "paths" ) )
+{
+    produces<nanoaod::FlatTable>();
+
+}
+
+  ~TrgBitTableProducer() override {}
+
+  void produce(edm::Event&, edm::EventSetup const&) override;
+
+
+  static void fillDescriptions(edm::ConfigurationDescriptions &descriptions) {}
+  
+
+private:
+
+  const edm::EDGetTokenT< edm::TriggerResults > trgresultsToken_;
+  // l1 seeds not implemented yet but can be added with litle effort
```

Comment:
I would add also the l1 seeds, so we can replace the current tables with this slimmer version.